apps/main/src/entities/gauge/model/query-options.ts

```diff
@@ -10,54 +10,58 @@
  * They encapsulate the data fetching logic, making it easier to manage and reuse across the application.
  */
 
-import { REFRESH_INTERVAL } from '@/constants'
+import { queryOptions } from '@tanstack/react-query'
 import * as api from '@/entities/gauge/api'
-import * as conditions from '@/entities/gauge/model/enabled-conditions'
 import { gaugeKeys as keys } from '@/entities/gauge/model'
-import type { DepositRewardApproveParams, GaugeQueryParams } from '@/entities/gauge/types'
-import { queryOptions } from '@tanstack/react-query'
+import * as conditions from '@/entities/gauge/model/enabled-conditions'
+import { DepositRewardApproveParams } from '@/entities/gauge/types'
+import { poolValidationGroup } from '@/entities/pool'
+import { createValidationSuite } from '@/shared/lib/validation'
+import { queryFactory } from '@/shared/model/factory'
+import { GaugeParams } from '@/shared/model/root-keys'
+import { REFRESH_INTERVAL } from '@/constants'
 
-export const getGaugeStatusQueryOptions = (params: GaugeQueryParams) =>
-  queryOptions({
-    queryKey: keys.status(params),
-    queryFn: api.queryGaugeStatus,
-    staleTime: REFRESH_INTERVAL['5m'],
-    enabled: conditions.enabledGaugeStatus(params),
-  })
+export const gaugeStatus = queryFactory({
+  queryKey: ({ chainId, poolId }: GaugeParams) => ['chain', chainId, 'pool', poolId, 'gauge', 'status'] as const,
```

Comment:
How can we use this key from outside?
To pass it to `dependencies`  for example

Reply:
`gaugeStatus.queryKey({...})`

Reply:
It's [re-exported](https://github.com/curvefi/curve-frontend/blob/108fc68d87ce09855c08a80ad1810c0477f4bbbb/packages/curve-lib/src/shared/model/factory.ts#L43) from the [factory output](https://github.com/curvefi/curve-frontend/blob/108fc68d87ce09855c08a80ad1810c0477f4bbbb/packages/curve-lib/src/shared/types/factory.ts#L34)